### ISSUE TITLE
Support "catch all" http trigger route

### ISSUE THREAD

**Issue by shibayan** [Azure/azure-functions-openapi-extension#70]
Writing a route like `{*path}` to get all subsequent path parameters did not correctly output the OpenAPI definition.

`Microsoft.Azure.WebJobs.Extensions.OpenApi`: `0.5.1-preview`

### Repro code

```csharp
[FunctionName("CatchAll")]
[OpenApiOperation(operationId: "CatchAll", tags: new[] { "name" })]
[OpenApiParameter("path", In = ParameterLocation.Path, Type = typeof(string))]
[OpenApiResponseWithoutBody(HttpStatusCode.OK)]
public static ActionResult<SampleModel> CatchAll([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "catchall/{*path}")]
                                                 HttpRequest req,
                                                 string path)
{
    return new OkObjectResult(path);
}
```

### Actual behavior

![image](https://user-images.githubusercontent.com/1356444/111282780-b3c70580-8681-11eb-8c97-932852c4f3fa.png)

**Comment by justinyoo** [comment]
@shibayan Thanks for the issue! The actual function endpoint itself works fine, if you send a request directly to the endpoint like:

```bash
curl http://localhost:7071/api/catchall/hello/world
```

However, like you mentioned, I can confirm that it doesn't work on the UI as expected. I'm not sure it's the issue on my end or the UI's end.

Have you got the similar experience with Swagger UI on other application like ASP.NET Core Web API with Swashbuckle?

**Comment by shibayan** [comment]
@justinyoo 

I tried the same route with the ASP.NET Core Web API with Swashbuckle, and it worked. Looking at the generated openapi definition, it seems that "*" should not be included in the definition.

```csharp
[Route("api/[controller]")]
[ApiController]
public class ValuesController : ControllerBase
{
    [HttpGet("{*path}")]
    public ActionResult Get(string path)
    {
        return Ok(path);
    }
}
```

```json
{
  "openapi": "3.0.1",
  "info": {
    "title": "WebApplication6",
    "version": "v1"
  },
  "paths": {
    "/api/Values/{path}": {
      "get": {
        "tags": [
          "Values"
        ],
        "parameters": [
          {
            "name": "path",
            "in": "path",
            "required": true,
            "schema": {
              "type": "string",
              "nullable": true
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Success"
          }
        }
      }
    }
  },
  "components": { }
}
```

**Comment by justinyoo** [comment]
Thanks for confirming! Let me have a look.

**Comment by justinyoo** [comment]
@shibayan I looked at the issue. However, if the wildcard character is omitted, it actually changes the semantics of the endpoint. In other words, the following path template implies the `{*path}` can be `hello`, `hello/world` or `hello/world/lorem/ipsum`.

```text
/catchall/{*path}
```

But the following path template implies the `{path}` can only be `hello` or `world`, not `hello/world`

```text
/catchall/{path}
```

I can change the path template from `/catchall/{*path}` to `/catchall/{path}` so that the OpenAPI doc shows the latter but, like I said, it changes semantics. Even though I applied the change to Swagger UI, it doesn't work properly either. It should be the UI's issue, not ours.

![image](https://user-images.githubusercontent.com/1538528/113465217-44b52380-946d-11eb-9e55-ad81469921e2.png)

![image](https://user-images.githubusercontent.com/1538528/113465125-9f9a4b00-946c-11eb-9247-982ed674b9f2.png)

![image](https://user-images.githubusercontent.com/1538528/113465241-7ded9380-946d-11eb-9857-75acfc75edf6.png)

![image](https://user-images.githubusercontent.com/1538528/113465258-9e1d5280-946d-11eb-91ed-ea9e2b75eba5.png)

**Comment by shibayan** [comment]
@justinyoo Thanks for the reply. I don't think the URL encoding of parameters is a Swagger UI issue, so I don't think it needs to be addressed, but I think the "catch all" parameter not being included in the OpenAPI definition is another issue.

**Comment by justinyoo** [comment]
@shibayan Yeah, it's not something I can do on my end. If you [Google it](https://www.google.com/search?q=openapi+spec+wildcard+in+the+path), there are so many discussions around - from the spec wise to the implementation perspective.

I'm not too sure, we can get it through the `proxies.json` file. Let me try.
